### PR TITLE
Add KafkaConnect to the Prometheus ServiceMonitor example.

### PR DESCRIPTION
**Pull request by allanmoso**
### Type of change
- Enhancement / new feature

### Description
We've found that the ServiceMonitor configuration did not include Kafka Connect metrics.  The changes in this pull request allows Prometheus to scrape Kafka Connect metrics as well.

### Checklist
_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
Thanks for the PR. Can you fix the DCO signoff? It should be:

```
git commit --amend --signoff
git push --force-with-lease origin feature/kafka-connect-prometheus
```

When clock on _Details_ next to the DCO status in the status section it should give more details.

**Comment by allanmoso** [comment]
You're welcome.  I think the DCO signoff should be fixed now.

File: examples/metrics/prometheus-install/strimzi-service-monitor.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ metadata:
     app: strimzi
 spec:
   selector:
-    matchLabels:
-      strimzi.io/kind: Kafka
+    matchExpressions:
+      - {key: "strimzi.io/kind", operator: In, values: ["Kafka", "KafkaConnect"]}
```
Comment (scholzj) [inline examples/metrics/prometheus-install/strimzi-service-monitor.yaml:10]: Does it work when it is instead written as YAML like the rest of the file?

```suggestion
      - key: strimzi.io/kind
        operator: In
        values: 
          - Kafka
          - KafkaConnect
```

**Comment by ppatierno** [comment]
Thanks for raising this, I realized the same while working on the Kafka Bridge metrics and that Kafka Connect metrics were not working properly. I have no comments more than the one made by Jakub.

**Comment by scholzj** [comment]
@ppatierno So should the bridge be added there as well?

**Comment by ppatierno** [comment]
@scholzj for the bridge I used a PodMonitor otherwise it wasn't working right now but it's a good point, it should work and it's even more general. I will try and open a PR after this one will be merged.

**Comment by ppatierno** [comment]
I did some tests but bridge dashboard needs some changes as well so I would leave this PR just for adding KafkaConnect then having a new one for bridge.

File: examples/metrics/prometheus-install/strimzi-service-monitor.yaml (hunk 1 above)
Comment (allanmoso) [reply examples/metrics/prometheus-install/strimzi-service-monitor.yaml]: Yes, it works in YAML form. I went ahead and changed it.

**Comment by ppatierno** [comment]
@allanmoso can you fix the DCO, please? Your latest commit is not signed off. Thanks!

**Comment by allanmoso** [comment]
DCO should be fixed now. Sorry about that.

**Pull request merged**